lib/screens/groupscreen/group.dart

```diff
@@ -71,20 +73,29 @@ class _GroupPageState extends State<GroupPage> with AutomaticKeepAliveClientMixi
             appBar: AppBar(
               title: Text('Group Details'),
               actions: <Widget>[
-                FlatButton.icon(
-                  textColor: getVisibleColorOnPrimaryColor(context),
-                  icon: Icon(FontAwesomeIcons.signOutAlt),
-                  onPressed: () async {
-                    //Navigator.push(context, MaterialPageRoute(builder: (context) => RootScreen()));
-                    try {
-                      await _request.exitGroup();
-                      Navigator.pop(context);
-                    } catch (e) {
-                      print(e.toString());
-                    }
-                  },
-                  label: Text('Leave Group'),
-                )
+                buttonEnabled
+                    ? FlatButton.icon(
+                        textColor: getVisibleColorOnPrimaryColor(context),
+                        icon: Icon(FontAwesomeIcons.signOutAlt),
+                        onPressed: () async {
+                          setState(() {
+                            buttonEnabled = false;
+                          });
+                          try {
+                            await _request.exitGroup();
+                            Navigator.pop(context);
+                          } catch (e) {
+                            print(e.toString());
+                          }
+                        },
+                        label: Text('Leave Group'),
+                      )
+                    : FlatButton.icon(
+                        textColor: getVisibleColorOnPrimaryColor(context),
+                        icon: Icon(FontAwesomeIcons.signOutAlt),
+                        onPressed: null,
+                        label: Text('Leave Group'),
+                      )
```

Comment:
So this buttonEnabled is added so that user doesn't click on leave group rapidly 2 times ?
} catch (e) {
print(e.toString());
} if it fails. Then reenable the button
You can ignore set state here by having a code like this

```dart
appBar: AppBar(
              title: Text('Group Details'),
              actions: <Widget>[
                FlatButton.icon(
                  textColor: getVisibleColorOnPrimaryColor(context),
                  icon: Icon(FontAwesomeIcons.signOutAlt),
                  onPressed: () async {
                    if (buttonEnabled) {
                      try {
                        buttonEnabled = false;
                        await _request.exitGroup();
                        Navigator.pop(context);
                      } catch (e) {
                        buttonEnabled = true;
                        print(e.toString());
                      }
                    }
                  },
                  label: Text('Leave Group'),
                )
              ],
            ),
```